pydoctor/templatewriter/writer.py

```diff
@@ -95,3 +116,21 @@ def _writeDocsForOne(self, ob, fobj):
         self.written_pages += 1
         ob.system.progress('html', self.written_pages, self.total_pages, 'pages written')
         flattenToFile(fobj, page)
+
+    def _checkTemplatesV(self) -> None:
```

Comment:
It seems we have two strategies for handling template directories:
- this method will scan the directory for templates early
- pages load templates on demand

On-demand loading has some downsides:
- broken templates will only be reported when HTML writing starts, which can be quite some time after startup on large code bases
- we seem to be doing a lot of repeated loading of the same templates

I think it would be useful to introduce something like a `Template` and `TemplateCollection` class, where the latter reflects the contents of one specific templates directory. `TemplateFileLookup` could then be renamed to `TemplateLookup` and perform the lookup in one or more`TemplateCollection` objects.

A `TemplateCollection` would parse all its templates on creation, so any errors are reported as soon as possible. And it would keep the `Template` objects, so we don't have to parse the same files multiple times.